### DISCUSSION TITLE
tracing can't use `type` as field

### DISCUSSION THREAD

**Discussion by BratSinot** [tokio-rs/tracing#2198] [Q&A]
### Version
tracing = "0.1.35"
tracing-subscriber = { version = "0.3.1", default-features = false, features = ["smallvec", "fmt", "tracing-log", "env-filter", "json", "time"] }

### Description
Can't use `type` as field name. This code give output `r#type`:
```rust
use tracing::info;
use tracing_subscriber::fmt;

fn main() {
    fmt().json().flatten_event(true).init();

    info!(r#type = "foo", "bar");
}
```
```log
{"timestamp":"2022-07-01T09:31:30.528058Z","level":"INFO","message":"bar","r#type":"foo","target":"playground2"}
```

If we change `r#type` on `type` code doesn't build:
```rust
    info!(type = "foo", "bar");
```
```log
error: no rules expected the token `type`
 --> src/main.rs:7:5
  |
7 |     info!(type = "foo", "bar");
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
  |
  = note: this error originates in the macro `$crate::fieldset` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by hawkw** [comment]
Field names are matched as Rust identifiers, as discussed in the documentation [here](https://docs.rs/tracing/latest/tracing/#recording-fields). Keywords, such as `type`, `match`, or `fn`, are not valid identifiers. 

The "raw identifier" (`r#`) syntax is supported by Rust, but requires special handling by `tracing` subscribers to strip the `r#` if it is present, which it appears the JSON formatter does not currently support. This could be fixed.

As an alternative, though, consider using a quoted field name. This is matched as a string literal, rather than an identifier. For example,
```rust
info!("type" = "foo", "bar");
```
